src/util/goblintutil.ml

```diff
@@ -445,3 +445,25 @@ let signal_of_string = let open Sys in function
   | s -> failwith ("Unhandled signal " ^ s)
 
 let self_signal signal = Unix.kill (Unix.getpid ()) signal
+
+module LazyEval (M : sig
+  type t
+  type result
+  val eval : t -> result
+end) : sig
+  type t
+  val make : M.t -> t
+  val force : t -> M.result
+end = struct
+  type t = { user_data : M.t; mutable value : M.result option }
```

Comment:
I doubt this makes any real difference, so probably not worth changing, but `user_data` isn't actually needed after the proper `value` has been computed. So it could just be
```ocaml
type t = { mutable value : (M.t, M.result) Either.t }
```
where `Left` holds the argument before evaluation and `Right` holds the result of the evaluation.